test/common/upstream/health_checker_impl_test.cc

```diff
@@ -346,8 +372,12 @@ class HttpHealthCheckerImplTest : public testing::Test {
   NiceMock<Runtime::MockRandomGenerator> random_;
   std::list<uint32_t> connection_index_{};
   std::list<uint32_t> codec_index_{};
+  const static HostWithHealthCheckMap health_checker_map_;
```

Comment:
Why static?

Reply:
I want to provide a default value for this arg here: https://github.com/dio/envoy/blob/f63e400ad7e9d04df504fdfc182d7e644d86cc58/test/common/upstream/health_checker_impl_test.cc#L285. Tried to use literal `{}`, but caught by ASAN in this [build](https://circleci.com/gh/envoyproxy/envoy/38457) the relevant line: https://gist.github.com/dio/1d2b925862f439cd7216595660bd6b8c#file-asan-log-L50.

Reply:
I think it would be preferable to do method overloading here than using a default arg; we're a bit allergic to the use of static non-PoD in the Envoy code base due to static initializaiton fiasco; even though it's probably perfectly safe here, it would be cleaner to just do method overloading in this use case.

Reply:
Got it. Hopefully, f2b5a32 is fine.